gflownet/utils/batch.py

```diff
@@ -1062,6 +1066,48 @@ def is_valid(self) -> bool:
             return False
         return True
 
+    def traj_indices_are_consecutive(self) -> bool:
+        """
+        Returns True if the trajectory indices start from 0 and are consecutive; False
+        otherwise.
+        """
+        return set(self.trajectories) == set(np.arange(self.get_n_trajectories()))
+
+    def make_indices_consecutive(self):
+        """
+        Updates the trajectory indices such that they start from 0 and are consecutive.
+
+        Returns
+        -------
+        self
+        """
+        if self.traj_indices_are_consecutive():
+            return self
+        traj_indices_unique, indices = np.unique(self.traj_indices, return_index=True)
+        traj_indices_unique = np.array(traj_indices_unique)[np.sort(indices)]
+        traj_indices_map_dict = OrderedDict(
+            zip(
+                traj_indices_unique,
+                np.arange(len(self.trajectories)),
+            )
+        )
+        self.trajectories = OrderedDict(
+            {
+                traj_idx_new: self.trajectories.pop(traj_idx_old)
```

Comment:
This should be minor, but why use `pop` here (and below for envs) instead of simple access? It should be marginally slower for no benefit I think?
```
>>> timeit.timeit('d={i: i for i in range(1000)}; [d.pop(i) for i in range(1000)]', number=10000)
0.9229674290399998
>>> timeit.timeit('d={i: i for i in range(1000)}; [d[i] for i in range(1000)]', number=10000)
0.7814702009782195
```

Reply:
I have re-implemented the whole thing anyway, without using `.pop()`. See [here](https://github.com/alexhernandezgarcia/gflownet/pull/167/commits/b28ded1da22ddec29db00fba8e116feecae01100).